core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -149,12 +149,25 @@ class LogCleaner(val config: CleanerConfig,
   }
 
   /**
-   * For testing, a way to know when work has completed. This method blocks until the
+   * For testing, a way to know when work has completed. This method waits until the
    * cleaner has processed up to the given offset on the specified topic/partition
+   *
+   * @param topic The Topic to be cleaned
+   * @param part The partition of the topic to be cleaned
+   * @param offset The first dirty offset that the cleaner doesn't have to clean
+   * @param maxWaitMs The maximum time in ms to wait for cleaner
+   *
+   * @return A boolean indicating whether the work has completed before timeout
    */
-  def awaitCleaned(topic: String, part: Int, offset: Long, timeout: Long = 30000L): Unit = {
-    while(!cleanerManager.allCleanerCheckpoints.contains(TopicAndPartition(topic, part)))
-      Thread.sleep(10)
+  def awaitCleaned(topic: String, part: Int, offset: Long, maxWaitMs: Long = 60000L): Boolean = {
```

Comment:
I think the first problem is minor and it is OK to change it twice. But I agree it is a valid problem. In stead of following either of our approach, I think it is better to do `sleepInterval = 100` to make this an explicit configuration.

I am a bit confused by your comments on the 2nd problem... I had this implementation previously that doesn't have this problem:

```
  def awaitCleaned(topic: String, part: Int, offset: Long, maxWaitMs: Long = 60000L): Boolean = {
    def isCleaned = cleanerManager.allCleanerCheckpoints.get(TopicAndPartition(topic, part)).fold(false)(_ >= offset)
    var remainingWaitMs = maxWaitMs
    while (!isCleaned && remainingWaitMs > 0) {
      Thread.sleep(math.min(100, remainingWaitMs))
      remainingWaitMs = math.max(0, remainingWaitMs - 100)
    }
    return isCleaned
  }
```

Then you made the following comment:

```
Great, thanks. Sorry to nitpick, but you don't need the math.max in remainingWaitMs = ... since we check for > 0 (that is, it's OK for the value to be negative). Looks good otherwise.
```

Therefore I change the implementation to do `remainingWaitMs -= 100`. Actually I am fine with both solution. But now I am confused by your latest comment that "it is not OK to have remainingWaitMs < 0". Did I misunderstand your comment?